src/Microsoft.AspNet.Authentication.OAuthBearer/OAuthBearerAuthenticationMiddleware.cs

```diff
@@ -38,16 +34,6 @@ public class OAuthBearerAuthenticationMiddleware : AuthenticationMiddleware<OAut
             ConfigureOptions<OAuthBearerAuthenticationOptions> configureOptions)
             : base(next, options, loggerFactory, encoder, configureOptions)
         {
-            if (Options.Notifications == null)
-            {
-                Options.Notifications = new OAuthBearerAuthenticationNotifications();
```

Comment:
Done. I also reverted the same change for OIDC and added a check for the OAuth2 generic middleware.